### DISCUSSION TITLE
PreFetch for ref in setup

### DISCUSSION THREAD

**Discussion by romkhram** [quasarframework/quasar#12488] [CLI - SSR mode]
```
const getParams = async () => {
  let params = await api.get('***')
  return params
}

export default defineComponent({
  preFetch ({ currentRoute, ssrContext }) {
    *** getParams() and put response to params.value  ? ***
  },
  setup () {
    const params = ref({})

    return {
      params
    }
  }
})
```

Can I use PreFetch feature for fill "params" from async response (without store or ref declaration before setup)

**Comment by pikil** [comment]
Technically, you can create a variable right there (before export) and set a watcher on it in setup(), which would update ref() if all used correctly, if that makes sense

Another approach, you can try Pinia, which is a next-gen Vuex (most likely, Vuex5 would be using Pinia's approach anyway)
